compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1657,7 +1658,13 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         // Postpone the evaluation of constants whose substs depend on inference
         // variables
         if substs.has_infer_types_or_consts() {
-            debug!("substs have infer types or consts: {:?}", substs);
+            let ac = AbstractConst::new(self.tcx, unevaluated.shrink());
+            if let Ok(None) = ac {
+                let mut uv = unevaluated;
+                uv.substs = InternalSubsts::identity_for_item(self.tcx, uv.def.did);
+                let param_env = self.tcx.param_env(uv.def.did);
+                return self.tcx.const_eval_resolve_for_typeck(param_env, uv, span);
```

Comment:
we want to erase regions before evaluating the constant. For that you probably want to just reuse the current remainder of this function ^^